IdentityCore/src/oauth2/MSIDTokenResponse.m

```diff
@@ -152,7 +152,7 @@ - (instancetype)initWithJSONDictionary:(NSDictionary *)json error:(NSError **)er
         _state = [json msidStringObjectForKey:MSID_OAUTH2_STATE];
         [self setIdToken:[json msidStringObjectForKey:MSID_OAUTH2_ID_TOKEN]];
         _error = [json msidStringObjectForKey:MSID_OAUTH2_ERROR];
-        _errorDescription = [json msidStringObjectForKey:MSID_OAUTH2_ERROR_DESCRIPTION];
+        _errorDescription = [[json msidStringObjectForKey:MSID_OAUTH2_ERROR_DESCRIPTION] msidURLDecode];
```

Comment:
can you ensure this will not impact other flows?

Reply:
Added unit test